packages/yarnpkg-core/sources/formatUtils.ts

```diff
@@ -236,7 +247,7 @@ export function pretty<T extends Type>(configuration: Configuration, value: Sour
 
   if (Object.prototype.hasOwnProperty.call(transforms, formatType)) {
     miscUtils.overrideType<keyof AllTransforms>(formatType);
-    return transforms[formatType].pretty(configuration, value as any);
+    return transforms[formatType].pretty(configuration, value as never);
```

Comment:
Types broke when the `filePath` argument above became stringlike

Reply:
I don't understand ... why does `never` work? 😮

Reply:
`transforms[formatType].pretty` expects the value to be `never` when `filePath` is stringlike, with the `as any` this is the type error
```
Argument of type 'any' is not assignable to parameter of type 'never'.
```

Reply:
Yep, but `never` shouldn't be there at all, since no callbacks in `transform` use it 🤔